pkg/storage/chunk/cache/memcached.go

```diff
@@ -222,6 +248,10 @@ func (c *Memcached) fetchKeysBatched(ctx context.Context, keys []string) (found
 		}
 	}
 
+	if workerErr != nil {
```

Comment:
This code block is not covered by a test

Reply:
This is the error returned when context cancellation happened in this method.

And should be captured in these test asserts.

```
				require.ErrorIs(t, err, context.Canceled)
```

and

```
				require.ErrorIs(t, err, context.DeadlineExceeded)
```

Reply:
I ran the tests with coverage and it showed this code block was not covered

Reply:
Also double checked by removing those lines (these asserts would fail in that case)

Reply:
Ah. I think I know why. I accidentially removed one of the return when it's closed. Caught in another test case. Fixing it.